index.js

```diff
@@ -1,79 +1,171 @@
+// TODO setState and getState
 const TAX = 0.08;
 const state = {
+  stepNumber: 0,
   customerBankAccount: 0,
   customerSpendLimit: 0,
-  phoneName: '',
-  phonePrice: '',
-  accessoryName: '',
-  accessoryPrice: '',
-  bundleSize: '',
+  phoneName: null,
+  phonePrice: 0,
+  accessoryName: null,
+  accessoryPrice: 0,
+  bundleSize: 0,
   totalSpent: 0,
 };
 
 initialize();
 
 function initialize() {
-  doneWithStepEl(1).onclick = () => { goToStep(2); };
-  doneWithStepEl(2).onclick = () => { goToStep(3); };
-  doneWithStepEl(3).onclick = () => { goToStep(4); };
+  goNextStep();
 
-  document.getElementById('back-step3').onclick = () => { goToPreviousStep(3); };
-  document.getElementById('back-step4').onclick = () => { goToPreviousStep(4); };
+  let count = 1;
+  while (count <= 4) {
+    nextButtonEl(count).onclick = () => { goNextStep(); };
+    count += 1;
+  }
 
-  goToStep(1);
+  backButtonEl(3).onclick = () => { goToPreviousStep(); };
+  backButtonEl(4).onclick = () => { goToPreviousStep(); };
 }
 
-function goToStep(stepNum) {
-  if (stepNum === 1 || stepNum === 2) {
-    toggleDisplaySteps(stepNum, (stepNum - 1));
-  } else if (stepNum === 3) {
-    // TODO - set validation for 0 or NaN
-    if (state.customerBankAccount === '' || state.customerSpendLimit === '') {
-      alert('Please fill out both inputs');
+
+function goNextStep() {
+  if (state.stepNumber <= 1) {
+    toggleDisplaySteps();
+  } else {
+    validateStep() && toggleDisplaySteps();
+  }
+}
+
+function validateStep() {
+  if (state.stepNumber === 2) {
+    if (state.customerBankAccount === 0 || Number.isNaN(state.customerBankAccount)
```

Comment:
It seems like `customerBankAccount` and `customerSpendLimit` have to be a number above 0, so you could forego all of this and just say:

```js
if (state.customerBankAccount && state.customerSpendLimit) {
  // ...
}
```

Reply:
Basically I'm targeting a bug with it. I'm not quite sure how to pull it off without `NaN` being specified.

`=== 0` is for the first time through, `isNaN` is for if the user returns to the page and empties the input. If they do, when state updates to a blank input, "NaN" is inserted into the innerHTML for either or both inputs in the top-bar. This prevents it from happening so I can force a blank innerHTML instead of `NaN`.

Removing the `=== 0` caused some issues too. I'll give it some more goes to see what I can come up with.

Reply:
Oh, sorry, I reversed the condition - if you add a logical NOT operator (`!`) before each of those, it will work: 

```js
if (state.customerBankAccount && state.customerSpendLimit) {
  // ...
}
```

Both `0` and `NaN` are falsy values, along with empty strings, `null`, `undefined` and `false`, so simply checking if a value is falsy (`!value`) is enough to catch either `0` or `NaN`.

Reply:
Got em